// react-speed-dial/dist/components/speed-dial/speed-dial.prop-types.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _propTypes = require('prop-types');

exports.default = {
	children: _propTypes.PropTypes.any,
	className: _propTypes.PropTypes.string,
	classNameBackdrop: _propTypes.PropTypes.string,
	classNameButtonWrap: _propTypes.PropTypes.string,
	classNameInTransition: _propTypes.PropTypes.string,
	classNameOpen: _propTypes.PropTypes.string,
	closeOnSecondClick: _propTypes.PropTypes.bool,
	closeOnScrollDown: _propTypes.PropTypes.bool,
	closeOnScrollUp: _propTypes.PropTypes.bool,
	enableMouseActions: _propTypes.PropTypes.bool,
	floatingActionButtonProps: _propTypes.PropTypes.shape({
		backgroundColor: _propTypes.PropTypes.string,
		className: _propTypes.PropTypes.string,
		disabled: _propTypes.PropTypes.bool,
		iconClassName: _propTypes.PropTypes.string,
		iconStyle: _propTypes.PropTypes.object,
		mini: _propTypes.PropTypes.bool,
		secondary: _propTypes.PropTypes.bool,
		style: _propTypes.PropTypes.object,
		zDepth: _propTypes.PropTypes.number
	}),
	hasBackdrop: _propTypes.PropTypes.bool,
	icon: _propTypes.PropTypes.object,
	iconOpen: _propTypes.PropTypes.object,
	isInitiallyOpen: _propTypes.PropTypes.bool,
	isOpen: _propTypes.PropTypes.bool,
	positionH: _propTypes.PropTypes.string,
	positionV: _propTypes.PropTypes.string,
	primaryText: _propTypes.PropTypes.string,
	style: _propTypes.PropTypes.object,
	styleBackdrop: _propTypes.PropTypes.object,
	styleButtonWrap: _propTypes.PropTypes.object,
	tabIndex: _propTypes.PropTypes.oneOfType([_propTypes.PropTypes.string, _propTypes.PropTypes.number]),
	toolbox: _propTypes.PropTypes.shape({
		height: _propTypes.PropTypes.number.isRequired,
		className: _propTypes.PropTypes.string,
		classNameMorphButton: _propTypes.PropTypes.string
	}),
	onClickPrimaryButton: _propTypes.PropTypes.func,
	onChange: _propTypes.PropTypes.func,
	onChangeState: _propTypes.PropTypes.func
};
